### PR TITLE
allow specifying compatible_installer per os

### LINKED ISSUE #2509: allow specifying minimum required installer version
to avoid having to do workarounds like [this](https://github.com/ubports/installer-configs/blob/master/v2/devices/axolotl.yml#L7), if the schema gets extended. should be a semver expression specifiable per operating system.

### PR DESCRIPTION
**Pull request by NeoTheThird**
resolves #2509

### CODE REVIEW

**Comment by NeoTheThird** [comment]
Don't worry about codecov there, mainevent was not tested yet previously and starting that now is out of scope for this

File: src/lib/mainEvent.js [hunk 1]
```diff
@@ -147,6 +147,17 @@ mainEvent.on("user:unlock", (fields, resolve) => {
   }).then(resolve);
 });
 
+// installer_version
+mainEvent.on("user:installer_version", (version, resolve) => {
+  log.warn(`Incompatible UBports Installer version! Required: ${version}`);
+  prompt({
+    title: "Incompatible installer version",
```
Comment (Flohack74) [inline src/lib/mainEvent.js:154]: Is this codecov warning a false positive? How could these lines not be covered but all others?

Comment (NeoTheThird) [reply src/lib/mainEvent.js:154]: the registration of the eventListener (calling `mainEvent.on()`) is executed, the actual listener isn't. The object literal passed to `prompt()` is evaluated statically, but never executed. The `.then()` function in never called thusly. CodeCov is correct, but the warning doesn't matter. we're not properly testing `mainEvent.js` yet.

Comment (Flohack74) [reply src/lib/mainEvent.js:154]: LOL did not see your comment in the first place :P

**Pull request merged**